src/events/event_processor.erl

```diff
@@ -0,0 +1,120 @@
+-module(event_processor).
+
+-export([
+    process/2,
+    async_process/2
+]).
+
+async_process(Vertex, Event) ->
+    spawn(?MODULE, process, [Vertex, Event]).
+    
+process(Vertex, Event) ->
+    base_instrumentation:event_started(Vertex, Event),
+    #{<<"type">> := Type} = Event,
+    case Type of
+        <<"child/updated">> -> process_child_change(Vertex, Event);
+        <<"child/removed">> -> process_child_removed(Vertex, Event);
+        <<"parent/updated">> -> process_parent_change(Vertex, Event);
+        <<"parent/removed">> -> process_parent_removed(Vertex, Event)
+    end,
+    base_instrumentation:event_finished(Vertex, Event),
+    inbox:free_vertex(Vertex),
```

Comment:
That is dangerous. Wrap it in try-catch and ALWAYS free vertex, regardless of success of the operation.